armi/reactor/reactors.py

```diff
@@ -1983,23 +1983,27 @@ def findAllAxialMeshPoints(self, assems=None, applySubMesh=True):
 
     def updateAxialMesh(self):
         """
-        Update axial mesh based on perturbed meshes of all the assemblies.
+        Update axial mesh based on perturbed meshes of the assemblies that are linked to the ref assem.
 
         Notes
         -----
-        While processLoading finds all axial mesh points, this method simply updates the values of the
-        known mesh with the current assembly heights. This does not change the number of mesh points.
+        While processLoading finds *all* axial mesh points, this method only updates the values of the
+        known mesh with the current assembly heights. **This does not change the number of mesh points**.
 
-        If `detailedAxialExpansion` is active, the global axial mesh param still only tracks the refAssem.
+        If ``detailedAxialExpansion`` is active, the global axial mesh param still only tracks the refAssem.
         Otherwise, thousands upon thousands of mesh points would get created.
+
+        See Also
+        --------
+        processLoading : sets up the master mesh that this perturbs.
         """
         # most of the time, we want fuel, but they should mostly have the same number of blocks
         # if this becomes a problem, we might find either the
         #  1. mode: (len(a) for a in self).mode(), or
         #  2. max: max(len(a) for a in self)
         # depending on what makes the most sense
         refAssem = self.refAssem
-
+        refMesh = self.findAllAxialMeshPoints([refAssem])
```

Comment:
This is fine for now, but i honestly think we should consider a moratorium on `refAssem` until we figure out how we really want to handle it.

Reply:
sure but this just pulls it out of a loop.